Deployment to sandbox failed.

```
Terraform will perform the following actions:

  # aws_iam_policy.gitlab_iam will be updated in-place
  ~ resource "aws_iam_policy" "gitlab_iam" {
        id        = "arn:aws:iam::122796619775:policy/azul-gitlab-iam"
        name      = "azul-gitlab-iam"
      ~ policy    = jsonencode(
          ~ {
              ~ Statement = [
                    # (13 unchanged elements hidden)
                    {
                        Action   = [
                            "kms:UpdateKeyDescription",
                            "kms:UpdateAlias",
                            "kms:UntagResource",
                            "kms:TagResource",
                            "kms:ScheduleKeyDeletion",
                            "kms:DeleteAlias",
                            "kms:CreateAlias",
                        ]
                        Effect   = "Allow"
                        Resource = [
                            "arn:aws:kms:us-east-1:122796619775:key/*",
                            "arn:aws:kms:us-east-1:122796619775:alias/*",
                        ]
                    },
                  + {
                      + Action   = "s3:GetObjects3:HeadBuckets3:HeadObject"
                      + Effect   = "Allow"
                      + Resource = "arn:aws:s3:::amazon-ssm-packages-us-east-1/*\",arn:aws:s3:::aws-ssm-document-attachments-us-east-1/*\","
                    },
                ]
                # (1 unchanged attribute hidden)
            }
        )
        tags      = {
            "Name"                = "azul-gitlab_iam"
            "billing"             = "hca"
            "component"           = "azul-gitlab_iam"
            "deployment"          = "dev"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (4 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_iam_policy.gitlab_iam: Modifying... [id=arn:aws:iam::122796619775:policy/azul-gitlab-iam]
╷
│ Error: updating IAM Policy (arn:aws:iam::122796619775:policy/azul-gitlab-iam): MalformedPolicyDocument: Actions/Condition can contain only one colon.
│ 	status code: 400, request id: 903c0f2d-ccd9-4bd8-950d-ed1775db9c7a
│
│   with aws_iam_policy.gitlab_iam,
│   on gitlab.tf.json line 1774, in resource[28].aws_iam_policy[0].gitlab_iam:
│ 1774:                     }
│
╵
make: *** [apply] Error 1
```

--- a/UPGRADING.rst
+++ b/UPGRADING.rst
@@ -20,6 +20,16 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
 have too many entries in this file.
 
 
+#5964 SSM Agent in GitLab lacks sufficient permissions in its role
+==================================================================
+
+Operator
+~~~~~~~~
+
+Manually deploy the ``gitlab`` component of any main deployment just before
+pushing the merge commit to the GitLab instance in that deployment.
+
+
 #5984 AWS event GetMacieSession results in AccessDenied error again
 ===================================================================
 

--- a/terraform/gitlab/gitlab.tf.json.template.py
+++ b/terraform/gitlab/gitlab.tf.json.template.py
@@ -938,6 +938,22 @@ emit_tf({} if config.terraform_component != 'gitlab' else {
                             f'arn:aws:kms:{aws.region_name}:{aws.account}:key/*',
                             f'arn:aws:kms:{aws.region_name}:{aws.account}:alias/*'
                         ]
+                    },
+
+                    # The SSM agent requires explicit permission to access
+                    # certain AWS-managed buckets if S3 traffic is routed
+                    # through a VPC gateway endpoint. See:
+                    # https://docs.aws.amazon.com/systems-manager/latest/userguide/ssm-agent-minimum-s3-permissions.html
+                    {
+                        'actions': [
+                            's3:GetObject',
+                            's3:HeadBucket',
+                            's3:HeadObject'
+                        ],
+                        'resources': [
+                            f'arn:aws:s3:::amazon-ssm-packages-{aws.region_name}/*',
+                            f'arn:aws:s3:::aws-ssm-document-attachments-{aws.region_name}/*'
+                        ]
                     }
                 ]
             }

